### PR TITLE
Review and Confirm page for Unit Note filing 

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* 
- bcgov/entity#16509

*Description of changes:*
- Review and Confirm page Unit Note filing
- Update some existing components to make them usable on the Review page
- Add some validations (not all)
- Add unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/parties/party/PartyAutocomplete.vue
Comment (dimak1) [inline ppr-ui/src/components/parties/party/PartyAutocomplete.vue:186]: Emit autocompleted Party all the way up to the ContactInformation component, and then set the store.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1401-0qbbbtml.web.app

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: Do we need the div here?

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: Looks good! 
Is this simple enough to be a basic reusable component?

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/UnitNoteReview.vue:9]: I'm not sure `mt-15` is a valid helper class via vuetify, is it providing a margin here?

Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:9]: Scratch that, just looked it up. Goes up to 16 now! Resolving.

Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:9]: Yea, it gave a margin of 60px. I can use our own class as well.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: I asked Liz to provide additional design samples, and after all variations are established, I can put into own component.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/UnitNoteReview.vue:198]: Its kind of a bummer this code is required in all instances of Staff Payment. 
After we upgrade, we should create a shim to consolidate all of this work, so all we need is to instantiate the component.

File: ppr-ui/src/store/state/state-model.ts
Comment (cameron-eyds) [inline ppr-ui/src/store/state/state-model.ts]: Is this required in the model? Shouldn't this be on our stateModel interface?

File: ppr-ui/src/store/state/state-model.ts
Comment (cameron-eyds) [inline ppr-ui/src/store/state/state-model.ts]: Same as above

**Review by cameron-eyds — APPROVED** [review]
Looks good, just some minor questions and comments.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:198]: Yea I know... I didn't want to bring it in here either - it really should be part of Staff Payment. Maybe a have our own wrapper?

File: ppr-ui/src/store/state/state-model.ts
Comment (dimak1) [reply ppr-ui/src/store/state/state-model.ts]: You mean the use of IF here? I will remove.

Comment (cameron-eyds) [reply ppr-ui/src/store/state/state-model.ts]: Yessir! Gracias

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:198]: Definitely, not part of your scope here just thinking its a future-us problem to tackle this. 
Probably post upgrade, we can consolidate these various instances of StaffPayment

File: ppr-ui/src/components/common/CertifyInformation.vue
Comment (orelbn) [inline ppr-ui/src/components/common/CertifyInformation.vue:133]: I think we can decide on the infoText in the script rather than in the template.

Can do something like content?.description || ...

File: ppr-ui/src/components/common/ContactInformation.vue
Comment (orelbn) [inline ppr-ui/src/components/common/ContactInformation.vue:7]: I have been using `showSomething`, instead of `!hideSomething`, so I think we should decide on a convention.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (orelbn) [inline ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: We have empty class tag here.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (orelbn) [inline ppr-ui/src/components/unitNotes/UnitNoteReview.vue:34]: There is inconsistencies with the margins on different flows. Maybe something worth bringing up with the designers. I already put a section tag and margin-top on the attention component itself, but now we need to add another one, because of the inconsistencies.

image

Or Maybe we can just put the section tag with the margin outside of the component in all cases.

**Review by orelbn — APPROVED** [review]
Looks awesome! Way to finish this so quickly 👍. A couple small comments.

**Review by doug-lovett — APPROVED** [review]
Looks good.

File: ppr-ui/src/components/unitNotes/UnitNoteReviewDetailsTable.vue
Comment (orelbn) [inline ppr-ui/src/components/unitNotes/UnitNoteReviewDetailsTable.vue]: Empty class tag here.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:34]: I would prefer not to have any (top/bottom) margins in the components themselves. Because of the inconsistencies that you mentioned, we should let the parent define the margins for each component where it's used.

Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:34]: Also, I'm not using the Folio component.

File: ppr-ui/src/components/common/ContactInformation.vue
Comment (dimak1) [reply ppr-ui/src/components/common/ContactInformation.vue:7]: The reason I made it like this, so that by default we have it always showing, and then simply pass `hidePartySearch`.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (orelbn) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:34]: I am aware, that was just to demonstrate the margin differences, its the same for attention in my flow.

I can update the attention in my current PR, to remove the margin and set it in the parent..

**Comment by dimak1** [comment]
@cameron-eyds @orelbn addressed your comments in latest commit.

File: ppr-ui/src/components/common/CertifyInformation.vue
Comment (dimak1) [reply ppr-ui/src/components/common/CertifyInformation.vue:133]: Good point, updated.

File: ppr-ui/src/components/unitNotes/UnitNoteReviewDetailsTable.vue
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReviewDetailsTable.vue]: Removed.

File: ppr-ui/src/store/state/state-model.ts
Comment (dimak1) [reply ppr-ui/src/store/state/state-model.ts]: Removed.

File: ppr-ui/src/store/state/state-model.ts
Comment (dimak1) [reply ppr-ui/src/store/state/state-model.ts]: Removed.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: Created own component for the Unit Notes details.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: Removed.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: Cool!

**Comment by cameron-eyds** [comment]
> @cameron-eyds @orelbn addressed your comments in latest commit.

Awesome, thanks! Looks great 👍

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1401-0qbbbtml.web.app

**Comment by orelbn** [comment]
Awesome looks great!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1401-0qbbbtml.web.app

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:198]: Yea we should clean up after the upgrade. I will create a ticket.

**Comment by dimak1** [comment]
Will merge after next release.

**Pull request merged**
